instance/tests/integration/base.py

```diff
@@ -40,10 +43,22 @@ def setUp(self):
         # Override the environment setting - always run task in the same process
         djhuey.HUEY.always_eager = True
 
+        # Use a reduced playbook for integration builds - it will run faster.
+        # See https://github.com/open-craft/configuration/blob/integration/playbooks/opencraft_integration.yml
+        patcher = patch.object(OpenEdXAppServer, 'CONFIGURATION_PLAYBOOK', new='opencraft_integration')
+        self.addCleanup(patcher.stop)
+        patcher.start()
+
     def tearDown(self):
-        OpenStackServer.objects.terminate()
-        for instance in SingleVMOpenEdXInstance.objects.iterator():
+        for appserver in OpenEdXAppServer.objects.iterator():
+            appserver.terminate_vm()
+        for instance in OpenEdXInstance.objects.iterator():
             instance.deprovision_swift()
             instance.deprovision_mongo()
             instance.deprovision_mysql()
         super().tearDown()
+
+        # All VMs should be terminated at this point, but check just in case:
+        if OpenStackServer.objects.exclude_terminated().count():
+            OpenStackServer.objects.terminate()
+            raise AssertionError("Some OpenStackServers were not terminated.")
```

Comment:
@bradenmacdonald Should the message mention that another attempt was made to terminate all VMs? Based on the current contents of the message a user might assume that they should find one or more VMs from the tests when checking OpenStack; but if the code in L63 above is successful in terminating remaining VMs, they wouldn't be able to and might get confused.